server/reload.go

```diff
@@ -41,6 +42,24 @@ func (d *debugOption) Apply(server *Server) {
 	server.Noticef("Reloaded: debug = %v", d.newValue)
 }
 
+// tlsOption implements the option interface for the `tls` setting.
+type tlsOption struct {
+	newValue *tls.Config
+}
+
+// Apply the tls change.
+func (t *tlsOption) Apply(server *Server) {
+	tlsRequired := t.newValue != nil
+	server.info.TLSRequired = tlsRequired
+	message := "disabled"
+	if tlsRequired {
+		server.info.TLSVerify = (t.newValue.ClientAuth == tls.RequireAndVerifyClientCert)
+		message = "enabled"
+	}
+	server.generateServerInfoJSON()
```

Comment:
I was about to ask to surround this with server lock/unlock but I realize that this is done higher in the caller stack. So we are good here.